include.sh

```diff
@@ -0,0 +1,130 @@
+fi
+export ENROOT_MOUNT_OPTIONS
+
+
+# Propagate these environment variables to the container
+PROPAGATED_ENV_VARIABLES=(BENCHMARK
+                          DRY_RUN
+                          EXECUTOR
+                          REPETITIONS
+                          SOLVER_REPETITIONS
+                          SEGMENTS
+                          SEGMENT_ID
+                          PRECONDS
+                          FORMATS
+                          ELL_IMBALANCE_LIMIT
+                          SOLVERS
+                          SOLVERS_PRECISION
+                          SOLVERS_MAX_ITERATIONS
+                          SOLVERS_GMRES_RESTART
+                          SYSTEM_NAME
+                          DEVICE_ID
+                          SOLVERS_JACOBI_MAX_BS
+                          BENCHMARK_PRECISION
+                          SOLVERS_RHS
+                          SOLVERS_RHS_FLAG
+                          SOLVERS_INITIAL_GUESS
+                          GPU_TIMER
+                          DETAILED
+                          MATRIX_LIST_FILE
+                          NVIDIA_VISIBLE_DEVICES
+                          CCACHE_DIR
+                          CCACHE_MAXSIZE
+                         )
+for bench_var in ${PROPAGATED_ENV_VARIABLES[*]}; do
+    check_var="CUSTOM_ENV_${bench_var}"
+    if [[ -n "${!check_var}" ]]; then
+        ENROOT_ENV_CONFIG+=("-e ${bench_var}=${!check_var}")
+    fi
+done
+export ENROOT_ENV_CONFIG
+
+# SLURM configuration variables.
+#
+# If the user sets any slurm variable or the variable USE_SLURM, this container
+# will use slurm job submission
+SUPPORTED_SLURM_VARIABLES=(SLURM_PARTITION
+                           SLURM_EXCLUSIVE
+                           SLURM_TIME
+                           SLURM_GRES
+                           SLURM_UPDATE_INTERVAL
+                           SLURM_PENDING_LIMIT
+                           SLURM_RUNNING_LIMIT
+                           USE_SLURM)
+for slurm_var in ${SUPPORTED_SLURM_VARIABLES[*]}; do
+    check_var="CUSTOM_ENV_${slurm_var}"
+    if [[ -n "${!check_var}" ]]; then
+        USE_SLURM=1
+    fi
+done
+export USE_SLURM
+# variables from slurm_utils we need to expose outside
+export SLURM_UPDATE_INTERVAL
+export SLURM_PENDING_LIMIT
+export SLURM_RUNNING_LIMIT
+export SLURM_GOOD_COMPLETED_STATUS
+export SLURM_GOOD_PENDING_STATUS
+export SLURM_BAD_STATUS
+
+
+function ensure_executable_available() {
+    local command=${1}
+
+    if ! type -p ${1} >/dev/null 2>/dev/null; then
+        die "No ${1} executable found"
+    fi
+}
+
+
+function logging() {
+    echo "==="
+    echo "= ${BASH_SOURCE[-1]}"
+    echo "==="
+    echo "Command line arguments: $@"
+    #if [[ $# -gt 0 ]]; then
+        # This is evil, delete the data afterwards
+        #echo "Script:"
+        #local SCRIPT="$1"
+        #if [[ -e "${SCRIPT}" ]]; then
+        #    cat "${SCRIPT}"
+        #fi
+    #fi
```

Comment:
also some option for logging?

Reply:
I removed this altogether as it was useful while debugging but there should be plenty of logging already. Coming up with a good logging strategy is also a bit more complex now: some scripts are executed directly, but some are also executed through slurm and the environment isn't the same, etc.